projects/applicationinsights-angularplugin-js/src/lib/applicationinsights-angularplugin-js.component.ts

```diff
@@ -186,7 +187,7 @@ export class AngularPlugin extends BaseTelemetryPlugin {
      * @param event The event that needs to be processed
      */
     processTelemetry(event: ITelemetryItem, itemCtx?: IProcessTelemetryContext) {
-        this.processNext(event, itemCtx);
+        runOutsideAngular(() => this.processNext(event, itemCtx));
```

Comment:
This one is likely not needed as this is called internally from all of the `trackXXX` calls -- so (if I'm following your comments correctly) this should already be getting run in the "outside" context...

Although, I do concede that if something does end up calling a track call within the Angular context (like any of the automatic events (JS errors, URL change etc)) then this would be the point at which it would switch to the different async context.

My only concern with this (which is less of an issue with any SPA framework) is that once the page starts to unload all operation "should" become synchronous to avoid the loss of events. And I don't think we current expose the internal flags indicating that we have received any unload signal (unload, beforeunload, pagehide and visibilitychange events)